datalad/tests/utils.py

```diff
@@ -1827,6 +1827,37 @@ def newfunc(*args, **kwargs):
     return newfunc
 
 
+def get_branch_commits(repo, branch=None, limit=None, stop=None):
+    """Return commit hexshas for a branch
```

Comment:
My main question is - why not to keep it as a method (well, rename to `get_branch_commits_`) of GitRepo and just remove `value` parameter, thus always yielding `hexsha`s?  
or we do not foresee any need for such functionality for anything but the tests?  I already see one usecase RFed above in https://github.com/datalad/datalad/pull/3834/files#diff-18445826672fc8f047d0b7d56512986cR327 for direct call to `git` (IMHO suboptimal to breed those)

Reply:
The idea here, and in other PRs before, is to get rid all these tiny wrapper around simple git calls and leave them to the return value focused wrappers we have now. I was under the impression that this direction had received general support. I dont see why this particular method should be treated differently.

Reply:
I have now implemented what you suggested @yarikoptic and adjusted the companion PR in the crawler too.